torch/_inductor/fx_passes/post_grad.py

```diff
@@ -789,20 +758,67 @@ def is_pointwise_use(use):
     return torch.Tag.pointwise in use.target.tags
 
 
-@register_graph_pattern(
-    CallFunction(aten.addmm, Arg(), Arg(), Arg()),
-    pass_dict=pass_patterns[2],
-)
-def unfuse_bias_add_to_pointwise(match: Match, inp, mat1, mat2):
+def should_prefer_unfused_addmm(match):
+    inp = match.kwargs["inp"]
     if not inp.meta["val"].is_cuda:
-        return
+        return False
 
     output = match.output_node()
-    if not all(is_pointwise_use(use) for use in output.users):
-        return
+    return all(is_pointwise_use(use) for use in output.users)
+
 
+@register_graph_pattern(
+    CallFunction(aten.addmm, KeywordArg("inp"), Arg(), Arg()),
+    pass_dict=pass_patterns[2],
+    extra_check=should_prefer_unfused_addmm,
+)
+def unfuse_bias_add_to_pointwise(match: Match, mat1, mat2, *, inp):
     def repl(inp, x1, x2):
         return x1 @ x2 + inp
 
     with V.fake_mode:
         match.replace_by_example(repl, [inp, mat1, mat2])
+
+
+def is_valid_addmm_fusion(match):
+    mat1, mat2 = match.args
+    inp = match.kwargs["inp"]
+
+    if not isinstance(inp, torch.fx.Node):
+        return False  # Input is a number
+
+    in_shape = inp.meta["val"].shape
+    matched = len(in_shape) <= 2
+    mm_shape = mat1.meta["val"].shape[0], mat2.meta["val"].shape[1]
+    for i, m in zip(in_shape, mm_shape):
+        matched &= i == 1 or i == m
```

Comment:
```suggestion
    matched = len(in_shape) <= 2 and all(i in (1, m) for i, m in zip(in_shape, mm_shape))
```

Now, what happens here if `in_shape` is size 1? You are comparing it just against shape[0]. Shouldn't you reverse it and zip it against `reverse(mm_shape)`?